x-pack/plugins/security_solution/server/endpoint/ingest_integration.ts

```diff
@@ -8,9 +8,63 @@ import { Logger } from '../../../../../src/core/server';
 import { NewPackageConfig } from '../../../ingest_manager/common/types/models';
 import { factory as policyConfigFactory } from '../../common/endpoint/models/policy_config';
 import { NewPolicyData } from '../../common/endpoint/types';
-import { ManifestManager, ManifestSnapshot } from './services/artifacts';
+import { ManifestManager } from './services/artifacts';
+import { Manifest } from './lib/artifacts';
 import { reportErrors, ManifestConstants } from './lib/artifacts/common';
-import { ManifestSchemaVersion } from '../../common/endpoint/schema/common';
+import { InternalArtifactSchema } from './schemas/artifacts';
+import { manifestDispatchSchema } from '../../common/endpoint/schema/manifest';
+
+const getManifest = async (logger: Logger, manifestManager: ManifestManager): Promise<Manifest> => {
+  let manifest: Manifest | null = null;
+
+  try {
+    manifest = await manifestManager.getLastComputedManifest(ManifestConstants.SCHEMA_VERSION);
+
+    // If we have not yet computed a manifest, then we have to do so now. This should only happen
+    // once.
+    if (manifest == null) {
+      // New computed manifest based on current state of exception list
+      const newManifest = await manifestManager.buildNewManifest(ManifestConstants.SCHEMA_VERSION);
+      const diffs = newManifest.diff(Manifest.getDefault(ManifestConstants.SCHEMA_VERSION));
+
+      // Compress new artifacts
+      const adds = diffs.filter((diff) => diff.type === 'add').map((diff) => diff.id);
+      for (const artifactId of adds) {
+        const compressError = await newManifest.compressArtifact(artifactId);
+        if (compressError) {
+          throw compressError;
+        }
+      }
+
+      // Persist new artifacts
+      const artifacts = adds
+        .map((artifactId) => newManifest.getArtifact(artifactId))
+        .filter((artifact) => artifact !== undefined) as InternalArtifactSchema[];
```

Comment:
Can the type be placed in the `map()` or `filter()` parameters instead? So something like:
```
adds.map((artifactId: InternalArtifactSchema) => {})
```

Reply:
@peluja1012 `artifactId` is a string... it's mapping `string => (InternalArtifactSchema | undefined)` ... The filter is removing the `undefined` instances, so I had to cast it to `InternalArtifactSchema[]` at the end... is there a better way?